src/main/java/org/opensearch/ad/feature/SearchFeatureDao.java

```diff
@@ -425,7 +426,7 @@ public void onResponse(SearchResponse response) {
                         listener.onResponse(topEntities);
                     } else if (expirationEpochMs < clock.millis()) {
                         if (topEntities.isEmpty()) {
-                            listener.onFailure(new IllegalStateException("timeout to get preview results.  Please retry later."));
+                            listener.onFailure(new AnomalyDetectionException("timeout to get preview results.  Please retry later."));
```

Comment:
In AD we have self-defined exceptions under class "AnomalyDetectionException" , this PR need to get concrete error message based on exception class, if it's "AnomalyDetectionException", we will get message from exception directly, otherwise we use general error message to avoid return verbose error message to user. Security team has concern about returning verbose error .